packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -1708,14 +1707,24 @@ class StackViewport extends Viewport implements IStackViewport {
       }
 
       function sendRequest(imageId, imageIdIndex, options) {
-        return loadAndCacheImage(imageId, options).then(
-          (image) => {
+        const uncompressedIterator = ProgressiveIterator.as(
+          loadAndCacheImage(imageId, options)
+        );
+        const displayedIterator = new ProgressiveIterator<void | IImage>(
+          'displayed'
+        );
+        displayedIterator.generate(async (it) => {
+          for await (const image of uncompressedIterator) {
+            // Update cache
+            cache.removeImageLoadObject(imageId);
+            cache.putImageLoadObject(imageId, {
+              promise: Promise.resolve(image),
+            });
             successCallback.call(this, image, imageIdIndex, imageId);
-          },
-          (error) => {
-            errorCallback.call(this, error, imageIdIndex, imageId);
+            it.add(image, uncompressedIterator.done);
           }
-        );
+        });
+        return displayedIterator.getDonePromise();
       }
 
```

Comment:
same comment as below that we should make this optionally happening based on a config, our old image loader should be default